`gb build` fails for cgo build w/ MinGW 4.9.2

Steps to reproduce my particular case (given that [GLFW 3.0](http://www.glfw.org) is installed and taken care of in the environment variables `CGO_CFLAGS` and `CGO_LDFLAGS`):

    $ mkdir -p demo/src && cd demo
    $ gb vendor fetch github.com/go-gl/glfw3/v3.0/glfw
    $ gb build github.com/go-gl/gflw3/v3.0/glfw

Seemingly, my `ld.exe` does not support the `--build-id` parameter, as the `gb build` eventually fails with

    c:/build/win_64/mingw-x86_64-4.9.2-release-win32-seh-rt_v3-rev0/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/4.9.2/../../../../x86_64-w64-mingw32/bin/ld.exe: unrecognized option '--build-id=none'
    c:/build/win_64/mingw-x86_64-4.9.2-release-win32-seh-rt_v3-rev0/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/4.9.2/../../../../x86_64-w64-mingw32/bin/ld.exe: use the --help option for usage information
    collect2.exe: error: ld returned 1 exit status
    FATAL command "build" failed: # gcc -fPIC -fmessage-length=0 -pthread -g -O2 -o C:\Users\username\AppData\Local\Temp\gb132194319\github.com\go-gl\glfw3\v3.0\_all.o C:\Users\username\AppData\Local\Temp\gb132194319\github.com\go-gl\glfw3\v3.0\_cgo_export.o C:\Users\username\AppData\Local\Temp\gb132194319\github.com\go-gl\glfw3\v3.0\error.o C:\Users\username\AppData\Local\Temp\gb132194319\github.com\go-gl\glfw3\v3.0\input.o C:\Users\username\AppData\Local\Temp\gb132194319\github.com\go-gl\glfw3\v3.0\monitor.o C:\Users\username\AppData\Local\Temp\gb132194319\github.com\go-gl\glfw3\v3.0\window.o C:\Users\username\AppData\Local\Temp\gb132194319\github.com\go-gl\glfw3\v3.0\clipboard.cgo2.o C:\Users\username\AppData\Local\Temp\gb132194319\github.com\go-gl\glfw3\v3.0\context.cgo2.o C:\Users\username\AppData\Local\Temp\gb132194319\github.com\go-gl\glfw3\v3.0\error.cgo2.o C:\Users\username\AppData\Local\Temp\gb132194319\github.com\go-gl\glfw3\v3.0\glfw.cgo2.o C:\Users\username\AppData\Local\Temp\gb132194319\github.com\go-gl\glfw3\v3.0\input.cgo2.o C:\Users\username\AppData\Local\Temp\gb132194319\github.com\go-gl\glfw3\v3.0\monitor.cgo2.o C:\Users\username\AppData\Local\Temp\gb132194319\github.com\go-gl\glfw3\v3.0\native_windows.cgo2.o C:\Users\username\AppData\Local\Temp\gb132194319\github.com\go-gl\glfw3\v3.0\time.cgo2.o C:\Users\username\AppData\Local\Temp\gb132194319\github.com\go-gl\glfw3\v3.0\util.cgo2.o C:\Users\username\AppData\Local\Temp\gb132194319\github.com\go-gl\glfw3\v3.0\window.cgo2.o -Wl,-r -nostdlib c:/externaldeps/build/win_64/mingw-x86_64-4.9.2-release-win32-seh-rt_v3-rev0/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/4.9.2/libgcc.a -Wl,--build-id=none: exit status 1

Building the package with `go get` works fine:

    $ go get github.com/go-gl/glfw3/v3.0/glfw && ls -l $GOPATH/pkg/windows_amd64/github.com/go-gl/glfw3/v3.0/glfw.a
    -rw-r--r-- 1 username Administratoren 467544 Sep  7 14:19 /c/Users/username/go/pkg/windows_amd64/github.com/go-gl/glfw3/v3.0/glfw.a

--- a/cgo.go
+++ b/cgo.go
@@ -277,7 +277,10 @@ func rungcc3(ctx *Context, dir string, ofile string, ofiles []string) error {
 		if err != nil {
 			return nil
 		}
-		args = append(args, libgcc, "-Wl,--build-id=none")
+		args = append(args, libgcc)
+	} else {
+		// explicitly disable build-id when using clang
+		args = append(args, "-Wl,--build-id=none")
 	}
 	t0 := time.Now()
 	err := run(dir, nil, gcc(), args...)

